pkg/bootstrapper/node_bootstrapper.go

```diff
@@ -10,38 +34,382 @@ package bootstrapper
+		StartType:    mgr.StartAutomatic,
+		ErrorControl: 0,
+		// Path to kubelet.exe
+		BinaryPathName:   filepath.Join(wcb.installDir, "kubelet.exe"),
+		LoadOrderGroup:   "",
+		TagId:            0,
+		Dependencies:     nil,
+		ServiceStartName: "",
+		DisplayName:      "",
+		Password:         "",
+		Description:      "",
+	}
+	wcb.kubeletSVC, err = wcb.svcMgr.CreateService(KubeletServiceName, filepath.Join(wcb.installDir, "kubelet.exe"), c, kubeletArgs...)
+	if err != nil {
+		return err
+	}
+	err = wcb.kubeletSVC.SetRecoveryActions([]mgr.RecoveryAction{
+		{Type: mgr.ServiceRestart, Delay: 5},
+	}, 600)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
-// TODO:  Add other functions as deemed necessary
+// startKubeletService starts the kubelet as a Windows service
+func (wcb *winNodeBootstrapper) startKubeletService() error {
+	if wcb.kubeletSVC == nil {
+		return fmt.Errorf("no kubelet service")
+	}
+	err := wcb.kubeletSVC.Start()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (wcb *winNodeBootstrapper) removeKubeletService() error {
+	if wcb.kubeletSVC == nil {
+		return nil
+	}
+	return wcb.kubeletSVC.Delete()
+}
+
+// controlService sends a signal to the service and waits until it changes state in response to the signal
+func (wcb *winNodeBootstrapper) controlService(c svc.Cmd, to svc.State) error {
+	status, err := wcb.kubeletSVC.Control(c)
+	if err != nil {
+		return err
+	}
+	// Most of the rest of the function borrowed from the package (golang.org/x/sys/windows/svc/mgr) example
+	// Arbitrary wait time
+	timeout := time.Now().Add(10 * time.Second)
+	for status.State != to {
+		if timeout.Before(time.Now()) {
+			return fmt.Errorf("timeout waiting for service to go to state=%d", to)
+		}
+		time.Sleep(300 * time.Millisecond)
+		status, err = wcb.kubeletSVC.Query()
+		if err != nil {
+			return fmt.Errorf("could not retrieve service status: %v", err)
+		}
+	}
+	return nil
+}
+
+func (wcb *winNodeBootstrapper) stopKubeletService() error {
+	if wcb.kubeletSVC == nil {
+		return nil
+	}
+	return wcb.controlService(svc.Stop, svc.Stopped)
+}
+
+// TODO: Remove OVN service as well
+// Stop stops and removes the kubelet service
+func (wcb *winNodeBootstrapper) Stop() error {
+	if wcb.kubeletSVC == nil {
+		return nil
+	}
+	wcb.stopKubeletService()
+	return wcb.removeKubeletService()
+
+}
+
+// refreshServiceManager will disconnect and reconnect from the Windows service API. In order to complete certain
+// operations, there must be zero handlers to the API present on the system.
+func (wcb *winNodeBootstrapper) refreshServiceManager() error {
+	var err error
+	if err = wcb.Disconnect(); err != nil {
+		return err
+	}
+	// We need to give Windows time to clean up the services we've marked for deletion
+	time.Sleep(5 * time.Second)
+	wcb.svcMgr, err = mgr.Connect()
+	return err
+}
+
+// TODO: add OVN service start here as well
+// Run sets up the install directory, creates the kubelet service, and then starts the kubelet service
```

Comment:
Why is this called `Run`? Is it because it is running the WMCB?

Reply:
Yes, added that to the comment